File: day_01/coding-examples/ruby/ex1.rb
class Ex1
  def run
    result = add2(2, 4)
    show_result(result)
  end

private

  def add2(n1, n2) 
    n1 + n2
  end

  def show_result(result)
    p result
  end
  # could  also be written like so:
  # def show_result(result); p result; end
end

# this will be executed since it is not enclosed in a function or other definition when the file is loaded
instance = Ex1.new
instance.run

File: day_01/coding-examples/javascript/ex1.js
function add2(n1, n2) { 
  return n1 + n2;
}

function showResult(result) {
  console.log(result);
}

// this will be executed since it is not enclosed in a function when the file is loaded
var result = add2(2, 4);
showResult(result);
